### ISSUE TITLE
SpriteFont Styles Not Working

### ISSUE THREAD

**Issue by borealwinter** [MonoGame/MonoGame#4149]
The SpriteFont does not display any styles other than regular;

**Comment by cra0zy** [comment]
Can confirm, will try to look into this tomorrow.

**Comment by repstyle** [comment]
I have the same issue, How can we fix it ?
VS 2013 . Monogame 3.4.

**Comment by tomspilman** [comment]
If I remember correctly it is because Freetype doesn't have the features that the Windows font system employs to generate them.  If you look at my post from 2014:

https://github.com/mono/MonoGame/issues/2741#issuecomment-51572666

> The bad news is that there is still one major issue... we don't support bold and italic styles. The Windows GDI will render a bold or italic font automatically, but FreeType has no such feature... so we need to invent one.

My understanding is that the Windows font system does the following when trying to resolve for a bold or italic font style:

 - First see if the TTF includes the style itself... few fonts do this.
 - Second look for another TTF in the same family that includes that style.
 - Lastly if all else fails Windows generates the style manually in its rendering engine.

Unfortunately Freetype does not have any features like this.  If we want these to work we would need to implement these three steps to get the same behavior in our font processor.

**Comment by repstyle** [comment]
Ok, i was using Verdana font (TrueType).
I have found a solution : Instead of using the style, i use "Verdana Bold" instead of "Veradana" for the font name.It's working.
So if you want to use styles for a font, in fact you will have to get a font for regular, one for bold, one for italic etc
It's the most clean and simple way to do it.

**Comment by tomspilman** [comment]
> It's the most clean and simple way to do it.

I agree...  just provide the right font name is a simple approach.  However people expect that setting a style works for fonts, so I think we still need to provide it.

**Comment by repstyle** [comment]
I totally agree with you.

**Comment by Tzenchor** [comment]
Just wanted to comment that i'm having a look at this.
I think I have had success with the option of looking for font files in the same family as the requested ones in Windows, but I need to test it a bit further.
About the option 1 (looking if the font has the style included), i'll need to read the font, but it feels wrong to use SharpFont outside `SharpFontImporter.cs`, so i'm not sure if I should move the font finding steps from `FontDescriptionProcessor.cs` to `SharpFontImporter.cs` or use SharpFont directly in `FontDescriptionProcessor.cs`.
Option 3 (Rendering the style manually) will be interesting to make italics

**Comment by wldevries** [comment]
This was fixed in the PR linked above and then #8570 caused a regression. This no longer works.

**Comment by ThomasFOG** [comment]
It appears that #8570 commented out your changes in ```MatchFont()``` from #7993 with the intent to implement it differently, but never did.
